Interop/StubGenerator/src/main/kotlin/org/jetbrains/kotlin/native/interop/gen/ObjCStubs.kt

```diff
@@ -256,18 +256,30 @@ internal fun Type.isStret(target: KonanTarget): Boolean {
         KonanTarget.MACOS_X64,
         KonanTarget.WATCHOS_X64,
         KonanTarget.TVOS_X64 -> when (unwrappedType) {
-            is RecordType -> unwrappedType.decl.def!!.size > 16 || this.hasUnalignedMembers()
+                is RecordType -> unwrappedType.decl.def!!.size > 16 || this.hasUnalignedMembers()
+                else -> false
+            }
+
+        // See: https://developer.apple.com/library/archive/documentation/DeveloperTools/Conceptual/LowLevelABI/130-IA-32_Function_Calling_Conventions/IA32.html#//apple_ref/doc/uid/TP40002492-SW5
+        // Structures 1 or 2 bytes in size are placed in EAX.
```

Comment:
Is it true for structures with float/double members?

Reply:
Yep.